editor/src/components/inspector/inspector-common.ts

```diff
@@ -655,8 +656,12 @@ export function detectFillHugFixedState(
   const isGroupLike = treatElementAsGroupLike(metadata, elementPath)
 
   const parsed = defaultEither(null, parseCSSLengthPercent(simpleAttribute))
-  if (parsed != null && parsed.unit === '%' && parsed.value === 100) {
-    const valueWithType: FixedHugFill = { type: isGroupLike ? 'hug-group' : 'fill', value: parsed }
+  if (parsed != null && parsed.unit === '%') {
+    const fillOrScaled = parsed.value === 100 ? 'fill' : 'scaled'
+    const valueWithType: FixedHugFill = {
+      type: isGroupLike ? 'hug-group' : fillOrScaled,
+      value: parsed,
```

Comment:
This works but tbh it's too dense for me, could you take this apart so that the logic that decides what the type will be is a little more explicit? (ideally into a helper function)